--- src/Chess/Board/MoveGen/Core.hs
+++ src/Chess/Board/MoveGen/Core.hs
@@ -180,3 +180,3 @@
                  r = ray kingSq attackerSq
                  -- Target mask: capture attacker or block ray
-                 targetMask = if r == 0 then bbFromSquare attackerSq else r
+                 targetMask = r .|. bbFromSquare attackerSq
                  -- Handle En Passant
@@ -263,3 +263,3 @@
              r = ray kingSq attackerSq
-             targetMask = if r == 0 then bbFromSquare attackerSq else r
+             targetMask = r .|. bbFromSquare attackerSq
          fillPawnEvasionPromotions b gs targetMask
